### PR TITLE
Fix `Style/IfUnlessModifier` bad precedence detection

### LINKED ISSUE #8282: Style/IfUnlessModifier bad precedence detection
The cop `Style/IfUnlessModifier` tests for being part of an assignment to add parenthesis, but that is way too naive. Being part of just about any operator expression won't work:

```
foo || if bar
        baz
      end
# autocorrected to:
foo || baz if bar
```

See also: #8281

### PR DESCRIPTION
**Pull request by tejasbubane**
Closes #8282 

-----------------

Before submitting the PR make sure the following are checked:

* [x] Wrote [good commit messages][1].
* [x] Commit message starts with `[Fix #issue-number]` (if the related issue exists).
* [x] Feature branch is up-to-date with `master` (if not - rebase it).
* [x] Squashed related commits together.
* [x] Added tests.
* [x] Added an entry to the [Changelog](https://github.com/rubocop-hq/rubocop/blob/master/CHANGELOG.md) if the new code introduces user-observable changes. See [changelog entry format](https://github.com/rubocop-hq/rubocop/blob/master/CONTRIBUTING.md#changelog-entry-format).
* [x] The PR relates to *only* one subject with a clear title and description in grammatically correct, complete sentences.
* [x] Run `bundle exec rake default`. It executes all tests and RuboCop for itself, and generates the documentation.

[1]: https://chris.beams.io/posts/git-commit/

### CODE REVIEW

File: spec/rubocop/cop/style/if_unless_modifier_spec.rb [hunk 1]
```diff
@@ -445,6 +445,32 @@ def f
     expect(corrected).to eq "a + (1 if b)\n"
   end
 
+  it 'adds parens in autocorrect when if-end used with OR operator' do
```
Comment (koic) [inline spec/rubocop/cop/style/if_unless_modifier_spec.rb:448]:
```suggestion
  it 'adds parens in autocorrect when if-end used with `||` operator' do
```

File: spec/rubocop/cop/style/if_unless_modifier_spec.rb [hunk 2]
```diff
@@ -445,6 +445,32 @@ def f
...
+        1
+      end
+    RUBY
+
+    expect_correction(<<~RUBY)
+      a || (1 if b)
+    RUBY
+  end
+
+  it 'adds parens in autocorrect when if-end used with AND operator' do
```
Comment (koic) [inline spec/rubocop/cop/style/if_unless_modifier_spec.rb:461]:
```suggestion
  it 'adds parens in autocorrect when if-end used with `&&` operator' do
```

**Comment by marcandre** [comment]
LGTM, after @koic's changes are dealt with. 👍

**Comment by koic** [comment]
Thanks!

**Pull request merged**